### PR TITLE
fix: improve debugging of applications in third party tools and Studio

### PR DESCRIPTION
**Pull request by feons**
https://jira.appcelerator.org/browse/ALOY-1612

### CODE REVIEW

File: Alloy/commands/compile/sourceMapper.js [hunk 1]
```diff
@@ -117,6 +120,8 @@ exports.generateCodeAndSourceMap = function(generator, compileConfig) {
 	// write the generated controller code
 	var outfile = target.filepath;
 	var relativeOutfile = path.relative(compileConfig.dir.project, outfile);
+	outputResult.code += `\n//# sourceMappingURL=file://${compileConfig.dir.project}/${CONST.DIR.MAP}/${relativeOutfile}.${CONST.FILE_EXT.MAP}`;
```
[the remark this answers is not in the document]
Comment (hansemannn) [reply Alloy/commands/compile/sourceMapper.js]: @feons Any update on that one? I've also requested @ewanharris for a QR as he's more into this 😙. Would be good to have source maps working for both iOS and Android.

**Comment by hansemannn** [comment]
@feons Can we finish this one?

**Comment by hansemannn** [comment]
@ewanharris I see activity here, whooop! :) Let me know if I can test on our larger scale project!

**Comment by nirmaljpatel** [comment]
Can one of you guide me on how this feature can be used? 

For a while I was able to use Safari Developer -> Simulator -> JSContext and see my JS files in the debugger. However, this doesn't seem to be working any more atleast for me with Ti SDK8.0.1.GA.

Is there a guide somewhere elaborating how to debug with Chrome/Safari dev-tools?

**Comment by ewanharris** [comment]
Hey @nirmaljpatel, this PR is currently going through some changes so I wouldn't recommend using it. We do have guides for debugging [iOS via Safari](https://docs.appcelerator.com/platform/latest/#!/guide/Debugging_iOS_Apps_with_3rd-party_Tools) and [Android via Chrome](https://docs.appcelerator.com/platform/latest/#!/guide/Debugging_Android_Apps_with_3rd-party_Tools), however be aware that for iOS there are currently some known issues [TIMOB-27098](https://jira.appcelerator.org/browse/TIMOB-27098) with debugging in Safari (which might be what you're referencing?)

**Comment by sgtcoolguy** [comment]
some open TODOs:
- We write a source map for `app/lib/` files externally, but we do not embed a `//# sourceMappingURL=` comment in the copied file to point at it. So tools like Babel or node-titanium-sdk don't "know" about the source map.
- The whole copying of `app/lib/` files is pretty hokey anyways. It copies the file, then we generate a 1:1 source map file (and in the process parse the file unnecessarily!), then after everything the "optimizing" step ends up running the babel transforms on the file when its doing it's sweep on the `Resources/` folder. The "copying" portion should do the transform and generate the source map and comment all at once.
- We should see if we can't just remove the special babel plugins/transforms altogether in Alloy (and just copy file contents with the source map stuff). I've started to try and move them to a babel plugin here: https://github.com/appcelerator/babel-plugin-transform-titanium that is intended to eventually be used by node-titanium-sdk's jsanalyze method. That way we can just have the same optimizations for all titanium builds and not have special "defines" or optimizations that only Alloy does.

**Pull request merged**